Buttplug.Server.Managers.ErosTek/ET312Protocol.cs

```diff
@@ -0,0 +1,579 @@
+            OCR1BL = 0x4048,                    // OCR1BL(IO Register)
+            OCR1BH = 0x4049,                    // OCR1BH(IO Register)
+            OCR1AL = 0x404a,                    // OCR1AL(IO Register)
+            OCR1AH = 0x404b,                    // OCR1AH(IO Register)
+            TCNT1L = 0x404c,                    // TCNT1L(IO Register)
+            TCNT1H = 0x404d,                    // TCNT1H(IO Register)
+            TCCR1B = 0x404e,                    // TCCR1B(IO Register)
+            TCCR1A = 0x404f,                    // TCCR1A(IO Register)
+            SFIOR = 0x4050,                     // SFIOR(IO Register)
+            OSCCAL_OCDR = 0x4051,               // OSCCAL/OCDR(IO Register)
+            TCNT0 = 0x4052,                     // TCNT0(IO Register)
+            TCCR0 = 0x4053,                     // TCCR0(IO Register)
+            MCUCSR = 0x4054,                    // MCUCSR(IO Register)
+            MCUCR = 0x4055,                     // MCUCR(IO Register)
+            TWCR = 0x4056,                      // TWCR(IO Register)
+            SPMCSR = 0x4057,                    // SPMCSR(IO Register)
+            TIFR = 0x4058,                      // TIFR(IO Register)
+            TIMSK = 0x4059,                     // TIMSK(IO Register)
+            GIFR = 0x405a,                      // GIFR(IO Register)
+            GICR = 0x405b,                      // GICR(IO Register)
+            OCR0 = 0x405c,                      // OCR0(IO Register)
+            SPL = 0x405d,                       // SPL(IO Register)
+            SPH = 0x405e,                       // SPH(IO Register)
+            SREG = 0x405f,                      // SREG(IO Register)
+            SenseOutputCurrent = 0x4060,        // ADC0: Output Current Sense COMM_MAIN_CBLOCK_BASE
+            SenseMAOffset = 0x4061,             // ADC1: Multi Adjust Offset - CBLOCK_MULTI_A_OFFSET
+            SensePSUVoltage = 0x4062,           // ADC2: Power Supply Voltage
+            SenseBatteryVoltage = 0x4063,       // ADC3: Battery Voltage
+            SenseLevelPotA = 0x4064,            // ADC4: Level Pot A - CBLOCK_POT_A_OFFSET
+            SenseLevelPotB = 0x4065,            // ADC5: Level Pot B - CBLOCK_POT_B_OFFSET
+            SenseAudioLevelA = 0x4066,          // ADC6: Audio Input Level A (Half wave)
+            SenseAudioLevelB = 0x4067,          // ADC7: Audio Input Level B (Half wave)
+            SenseButtons = 0x4068,              // Current pushed buttons
+            SenseButtonLatched = 0x4069,        // Last pushed buttons
+            MasterTimerMSB = 0x406A,            // Master timer (MSB) (0x4073 LSB) runs 1.91Hz
+            OutputCalibrationA = 0x406B,        // Channel A calibration (DAC power offset)
+            OutputCalibrationB = 0x406C,        // Channel B calibration (DAC power offset)
+            MenuState = 0x406D,                 // Menu State
+            Adr406e = 0x406E,                   // unused
```

Comment:
Unused const addresses like this (`AdrXXXX`) aren't needed and can be easily added later on. I recommend not including them unless they're explicitly reserved (which they're not in this cases)

Reply:
Ok, so this actually brings up a good question. Right now, we have the ET312 constants in their own manager. However, is there a point where we might want to expose these to clients for usage, i.e. if we end up with an ET312Cmd command that allows remote peek/poke? I'm not even sure how that would look yet because the ET312 is by far the most complicated piece of hardware we support now. Just throwing the idea out there. If we did want something like that, we might want to keep the full constant list around.

Reply:
they're cast to a `uint` when used in `poke()` and `peek()` so a developer supplying a raw value to the methods will remain functional in the meantime until a PR is submitted to associate a name with the value. Where as if someone were to reference the name `AdrXXXX` for a register and the name is changed later on, it'll be a breaking change since the name would no longer exist

Reply:
I've included the "dummy" constants mainly to preserve parity with the documentation, which also lists all variables used by the firmware internally, and I thought the commentary was worth preserving in the header file.

As of exposing those low-level memory operations in the buttplug protocol: Please don't. That would be like piping the low-level BLE commands of other toys through buttplug, only that in this case, that would open up serious safety / security issues. The user has already to trust us quite a bit by connecting their gonads to a box we control, giving every client (possibly over the network) the ability to p0wn the box without constraints is not a good idea, IMHO :)